.github/workflows/build_and_tests.yml

```diff
@@ -18,42 +18,40 @@ jobs:
     - name: Clone
       uses: actions/checkout@v2
     - name: Speculos dependencies
-      run: sudo apt-get update && sudo apt-get install -y qemu-user-static
+      run: sudo apt-get update && sudo apt-get install -y qemu-user-static gcc-arm-linux-gnueabihf libvncserver-dev
     - name: Build & install
       run: |
         pip install --extra-index-url https://test.pypi.org/simple/ -U .[tests,all_backends]
         pip install -U click>=8
-    - name: Gather boilerplate NanoS to run tests against
-      uses: dawidd6/action-download-artifact@v2
-      with:
-        repo: LedgerHQ/app-boilerplate
-        name: boilerplate-app-nanos
-        workflow: ci-workflow.yml
-        path: tests/elfs/
-    - name: Renaming NanoS artifact
-      run: mv tests/elfs/app.elf tests/elfs/nanos.elf
-    - name: Gather boilerplate NanoX to run tests against
-      uses: dawidd6/action-download-artifact@v2
+    - name: Clone Speculos
+      uses: actions/checkout@v3
       with:
-        repo: LedgerHQ/app-boilerplate
-        name: boilerplate-app-nanox
-        workflow: ci-workflow.yml
-        path: tests/elfs/
-    - name: Renaming NanoS artifact
-      run: mv tests/elfs/app.elf tests/elfs/nanox.elf
-    - name: Gather boilerplate NanoS+ to run tests against
+        repository: LedgerHQ/speculos
+        ref: improve_client_events
+        path: speculos
+    - name: Install Speculos 
+      run: |
+        cd speculos
+        pip install .
+    - name: Gather boilerplate for Nano S/X/S+ to run tests against
```

Comment:
I guess this changes was necessary due to recent changes in app-boilerplate?
Could you put in in another dedicated PR that we could merge quickly?

Reply:
Yes I will open a new PR once @fbeutin-ledger gives me the go. He needs to update the name of binary artifact archive to something more explicit than just "apps"